### ISSUE TITLE
No syntax highlighting after #81

### ISSUE THREAD

**Issue by justinpage** [leafgarland/typescript-vim#82]
## Problem

After merge of #81 there is no longer any syntax highlighting. Filetype defaults to `xml`.
## Idea

Revert back a commit and syntax highlighting works. Not sure if there is an alternative solution to the setlocal file type?

**Comment by leafgarland** [comment]
Hi, I don't see that problem. The change in #81 means that the addin is less aggressive in setting the filetype. Perhaps something else in your Vim setup is trying to set the filetype?

**Comment by leafgarland** [comment]
You could try this in a typescript buffer

```vim
:verbose set filetype
```

That might tell you what set the filetype to `xml`?

**Comment by leftiness** [comment]
` filetype=xml Last set from /usr/share/nvim/runtime/filetype.vim`

**Comment by heilaw** [comment]
Syntax highlighting is no longer working in #81 . 79e6e53 also confirmed working.

OS X 10.11.5 with vim 7.4 from Mac Ports.

**Comment by highlyunavailable** [comment]
Also having this issue.

```
:verbose set filetype returns:
  filetype=xml
        Last set from /usr/local/Cellar/macvim/7.4-101/MacVim.app/Contents/Resources/vim/runtime/filetype.vim
```

which contains:

```vim
" Qt Linguist translation source and Qt User Interface Files are XML
au BufNewFile,BufRead *.ts,*.ui			setf xml
```

**Comment by leftiness** [comment]
I'm not exactly sure what @lambdalisue from #81 was doing. I use git diff when I want a diff. I understand that filetype.vim is setting the type to XML because it thinks the \*.ts file is a Qt Linguist file, but I think it's wrong to comment out that section from the default. Should the plugin override the default? Should I put `autocmd BufNewFile,BufRead *.ts,*.tsx setlocal filetype=typescript` in my vimrc and accept that it will cause a problem if I'm trying to do whatever lamdalisue was trying to do?

**Comment by highlyunavailable** [comment]
I'd expect that plugins would overwrite defaults, yes.

The change in #81 made typescript "not override" the already set filetype: http://vimdoc.sourceforge.net/htmldoc/options.html#%3asetfiletype

All the diff modes that I've tried other than doing `diff 1.ts 2.ts > diff.ts` and opening `diff.ts` in VIM produced very usable diffs with proper highlighting  - I'm not sure what is going on that means we'd not want to override filetype for `.ts` files.

For now I've put `autocmd BufNewFile,BufRead *.ts,*.tsx setlocal filetype=typescript` in my vimrc and am accepting consequences (which I've seen none of).

**Comment by leafgarland** [comment]
Not sure what's going on here but thanks for all the bug reports. I have reverted the change from #81, and will investigate further. The docs that were mentioned in #81 do seem to suggest that using `setfiletype` is correct.

**Comment by lambdalisue** [comment]
Using `filetype.vim` instaed of `ftdetect/typescript.vim` would solve the problem while

>   **The files in the "ftdetect" directory are used after all the default
>   checks**, thus they can overrule a previously detected file type.  But you
>   can also use |:setfiletype| to keep a previously detected filetype.

>   **Your filetype.vim will be sourced before the default FileType autocommands
>   have been installed**.  Your autocommands will match first, and the
>   ":setfiletype" command will make sure that no other autocommands will set
>   'filetype' after this.

http://vimdoc.sourceforge.net/htmldoc/filetype.html#filetype

**Comment by lambdalisue** [comment]
I'm sorry for that. <del>I should have carefully used `filetype.vim` instead of `ftdetect/typescript.vim` I guess</del>.

As I mentioned in #81, using `filetype.vim` should not be performed on any plugins I guess so...

1. Remove `ftdetect/typescript.vim` while it does not work (personally prefer)
2. Ignore my suggestion

I personally perfer 1. while

- Adding `autocmd BufFileNew, ...` to user's vimrc is easy and configurable by user
- Removing `autocmd BufFileNew,...` from plugin is a bit difficult and not configurable by user so users would requires some workarounds for filetype detection.

But these are all depends on you.

**Comment by DABH** [comment]
Just a heads up (2018), I just clean installed Vundle and typescript-vim and on Linux and had to add `autocmd BufNewFile,BufRead *.ts,*.tsx setlocal filetype=typescript` (as suggested above) in order to see any syntax highlighting (vim was also detecting ts files as xml, as mentioned above).  Luckily I found this issue.  Can we just add that instruction to the README, either as an installation instruction or as an FAQ?  My environment has nothing special in it so I feel like this could potentially affect a lot of new users.  Thanks!

**Comment by DABH** [comment]
Thanks for the reply!

I'm using vim 8.0.707 on Debian stretch 9.4.  My vimrc looks like
```
set nocompatible              " be iMproved, required
syntax on

" set the runtime path to include Vundle and initialize
set rtp+=~/.vim/bundle/Vundle.vim
call vundle#begin()
" alternatively, pass a path where Vundle should install plugins
"call vundle#begin('~/some/path/here')

" let Vundle manage Vundle, required
Plugin 'VundleVim/Vundle.vim'

" The following are examples of different formats supported.
" Keep Plugin commands between vundle#begin/end.
" plugin on GitHub repo
" Plugin 'tpope/vim-fugitive'
" plugin from http://vim-scripts.org/vim/scripts.html
" Plugin 'L9'
" Git plugin not hosted on GitHub
" Plugin 'git://git.wincent.com/command-t.git'
" git repos on your local machine (i.e. when working on your own plugin)
" Plugin 'file:///home/gmarik/path/to/plugin'
" The sparkup vim script is in a subdirectory of this repo called vim.
" Pass the path to set the runtimepath properly.
" Plugin 'rstacruz/sparkup', {'rtp': 'vim/'}
" Install L9 and avoid a Naming conflict if you've already installed a
" different version somewhere else.
" Plugin 'ascenator/L9', {'name': 'newL9'}

Plugin 'leafgarland/typescript-vim'

" All of your Plugins must be added before the following line
call vundle#end()            " required
filetype plugin indent on    " required
" To ignore plugin indent changes, instead use:
"filetype plugin on
"
" Brief help
" :PluginList       - lists configured plugins
" :PluginInstall    - installs plugins; append `!` to update or just :PluginUpdate
" :PluginSearch foo - searches for foo; append `!` to refresh local cache
" :PluginClean      - confirms removal of unused plugins; append `!` to auto-approve removal
"
" see :h vundle for more details or wiki for FAQ
" Put your non-Plugin stuff after this line

autocmd BufNewFile,BufRead *.ts,*.tsx setlocal filetype=typescript

set tabstop=4 softtabstop=0 expandtab shiftwidth=4
syntax on
```

(I had to add that autocmd line to make the highlighting work, as mentioned.)  I don't think I did any fancy customizations beyond adding Vundle and typescript-vim -- but if you spot something wrong, just let me know.  Thanks again for taking a look!

**Comment by adyouri** [comment]
Hi, @DABH @leafgarland 

I had this same issue, as I understand it, this is caused by `syntax on` being before `call vundle#end()` (@DABH try removing the second line in your `vimrc`, since you have `syntax on` at the end of it).

By having only one `syntax on` line after the line `filetype plugin indent on    " required` I was able to get the syntax highlighting working without an `autocmd` line.

But I have another issue, `let` and `var` don't get highlighted as language keywords, but `const` does, which is pretty weird. I am new to TypeScript so I don't know if there are more keywords that don't get highlighted, @leafgarland any idea what is happening here?.

And Thanks to everyone who commented on this issue!

**Comment by asfaltboy** [comment]
Hi, I use [dein][1] and I have the same issue above, the workaround doesn't seem to do anything for me :( .

[1]: https://github.com/Shougo/dein.vim
